### PR TITLE
Generic L431C(B-C)(T-U) Variant

### PR DESCRIPTION
**Pull request by jordancormack**
Adding generic L431C(B-C)(T-U) board variants.

I followed the wiki guide, but this is the first new variant I've added, so hopefully I have done everything correctly!

### CODE REVIEW

File: variants/STM32L4xx/L431C(B-C)(T-U)/generic_clock.c [hunk 1]
```diff
@@ -21,8 +21,47 @@
...
+  RCC_OscInitStruct.PLL.PLLR = RCC_PLLR_DIV2;
+  if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK)
+  {
+    Error_Handler();
+  }
+
+  /** Initializes the CPU, AHB and APB buses clocks
+  */
+  RCC_ClkInitStruct.ClockType = RCC_CLOCKTYPE_HCLK|RCC_CLOCKTYPE_SYSCLK
+                              |RCC_CLOCKTYPE_PCLK1|RCC_CLOCKTYPE_PCLK2;
```
Comment (fpistm) [inline variants/STM32L4xx/L431C(B-C)(T-U)/generic_clock.c:55]:
```suggestion
  RCC_ClkInitStruct.ClockType = RCC_CLOCKTYPE_HCLK | RCC_CLOCKTYPE_SYSCLK
                                | RCC_CLOCKTYPE_PCLK1 | RCC_CLOCKTYPE_PCLK2;
```

File: README.md [hunk 2]
```diff
@@ -676,6 +676,7 @@ User can add a STM32 based board following this [wiki](https://github.com/stm32d
 | Status | Device(s) | Name | Release | Notes |
 | :----: | :-------: | ---- | :-----: | :---- |
 | :green_heart:  | STM32L412K8<br>STM32L412KB<br>STM32L422KB | Generic Board | *2.0.0* |  |
+| :yellow_heart:  | STM32L431CB<br>STM32L431CC | Generic Board |  |  |
```
Comment (fpistm) [inline README.md:679]:
```suggestion
| :yellow_heart:  | STM32L431CB<br>STM32L431CC | Generic Board | **2.8.1** |  |
```

**Review by fpistm — CHANGES_REQUESTED** [review]
Some minor changes to do and it will be ok for merge. Thanks.

File: variants/STM32L4xx/L431C(B-C)(T-U)/ldscript.ld [hunk 3]
```diff
@@ -46,7 +46,7 @@ _Min_Stack_Size = 0x400; /* required amount of stack */
 MEMORY
 {
   RAM    (xrw)    : ORIGIN = 0x20000000,   LENGTH = LD_MAX_DATA_SIZE
-  RAM2    (xrw)    : ORIGIN = 0x10000000,   LENGTH = 16K
+  RAM2    (xrw)    : ORIGIN = 0x2001C000,   LENGTH = 16K
```
Comment (fpistm) [inline variants/STM32L4xx/L431C(B-C)(T-U)/ldscript.ld:49]: No the value is correct. This was just a note :wink:
RAM2 is also mapped at the end of the RAM, that's all.
```suggestion
  RAM2    (xrw)    : ORIGIN = 0x10000000,   LENGTH = 16K
```

**Review by fpistm — CHANGES_REQUESTED** [review]

File: variants/STM32L4xx/L431C(B-C)(T-U)/generic_clock.c [hunk 4]
```diff
@@ -21,8 +21,44 @@
...
+  RCC_OscInitStruct.PLL.PLLQ = RCC_PLLQ_DIV2;
+  RCC_OscInitStruct.PLL.PLLR = RCC_PLLR_DIV2;
+  if (HAL_RCC_OscConfig(&RCC_OscInitStruct) != HAL_OK) {
+    Error_Handler();
+  }
+
+  /** Initializes the CPU, AHB and APB buses clocks
+  */
+  RCC_ClkInitStruct.ClockType = RCC_CLOCKTYPE_HCLK | RCC_CLOCKTYPE_SYSCLK
+                              | RCC_CLOCKTYPE_PCLK1 | RCC_CLOCKTYPE_PCLK2;
```
Comment (fpistm) [inline variants/STM32L4xx/L431C(B-C)(T-U)/generic_clock.c:53]:
```suggestion
                                | RCC_CLOCKTYPE_PCLK1 | RCC_CLOCKTYPE_PCLK2;
```

File: variants/STM32L4xx/L431C(B-C)(T-U)/ldscript.ld (hunk 3 above)
Comment (jordancormack) [reply variants/STM32L4xx/L431C(B-C)(T-U)/ldscript.ld]: ah okay, I thought I had seen it both ways in other variant ldscript.ld, so was not sure.

thanks

**Comment by jordancormack** [comment]
@fpistm anything else I need to do for this?

**Review by fpistm — APPROVED** [review]
LGTM

**Comment by fpistm** [comment]
> @fpistm anything else I need to do for this?

All is fine. I was OOO the last 4 days. 😉  Thanks for the PR.

**Pull request merged**